src/components/Menu/Menu/hooks/useSubMenuIndex.tsx

```diff
@@ -1,11 +1,11 @@
 import { useCallback, useState } from "react";
 
 export default function useSubMenuIndex() {
-  const [openSubMenuIndex, setOpenSubMenuIndex] = useState(null);
-  const hasOpenSubMenu = openSubMenuIndex || openSubMenuIndex === 0;
+  const [openSubMenuIndex, setOpenSubMenuIndex] = useState<number>(null);
+  const hasOpenSubMenu = !!openSubMenuIndex || openSubMenuIndex === 0;
```

Comment:
Unrelated to your PR, but this logic is a little bit messy and implicit, making it hard to understand what you're actually checking for here (which I'm assuming, whether that index is a number). How about something like:

```suggestion
  const hasOpenSubMenu = typeof openSubMenuIndex === 'number';
```